packages/x-components/src/composables/use-store.ts

```diff
@@ -1,13 +1,40 @@
-import { getCurrentInstance } from 'vue';
+import { computed, ComputedRef, getCurrentInstance } from 'vue';
 import { Store } from 'vuex';
+import { Dictionary } from '@empathyco/x-utils';
+import { ExtractState, XModuleName } from '../x-modules/x-modules.types';
 
 /**
- * Function which returns the `$store` object from the current component instance.
+ * Function which returns the `$store` object from the current component instance
+ * and the selected state as a dictionary of paths.
  *
- * @returns The `$store` object from the current component instance.
+ * @param module - The {@link XModuleName} of the getter.
+ * @returns The state properties of the module and the `$store`.
  *
  * @public
  */
-export function useStore(): Store<any> {
-  return (getCurrentInstance()?.proxy as { $store: Store<any> }).$store;
+export function useStore<Module extends XModuleName, Path extends keyof ExtractState<Module>>(
+  module?: Module
+): UseStore<Module, Path> {
+  const store = (getCurrentInstance()?.proxy as { $store: Store<any> }).$store;
+  const useState = (module: Module, paths: Path[]): ComputedRef<Dictionary<keyof Path>> => {
+    return paths.reduce<ComputedRef<Dictionary<keyof Path>>>((state, path) => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      state[path] = computed(() => store.state.x[module][path]);
+      return state;
+    }, {} as ComputedRef<Dictionary<keyof Path>>);
+  };
+  const useStateWithModule = useState.bind(module);
+  return {
+    store,
+    useState: module ? useStateWithModule : useState
+  };
 }
```

Comment:
The types for the useStore are wrong. We don't have a ComputedRef of a Dictionary but the other way around. It should be something like:

```ts
type UseStore = {
  store: Store<any>;
  useState: <Module extends XModuleName, Paths extends keyof ExtractState<Module>>(
    module: Module,
    paths: Paths[]
  ) => Dictionary<ComputedRef>;
};

export function useStore(): UseStore {
  const store = (getCurrentInstance()?.proxy as { $store: Store<any> }).$store;
  const useState = <Module extends XModuleName, Paths extends keyof ExtractState<Module>>(
    module: Module,
    paths: Paths[]
  ): Dictionary<ComputedRef> => {
    return paths.reduce<Dictionary<ComputedRef>>((state, path) => {
      state[path as string] = computed(() => store.state.x[module][path]);
      return state;
    }, {});
  };
  return {
    store,
    useState
  };
```

For this first iteration I would skip the use case of passing the module as argument.